src/Common/FailPoint.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+#include "config.h"
+
+#include <Common/Exception.h>
+#include <Core/Types.h>
+
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdocumentation"
+#pragma clang diagnostic ignored "-Wreserved-macro-identifier"
+#endif
+
+#include <fiu.h>
+#include <fiu-control.h>
+
+#ifdef __clang__
+#pragma clang diagnostic pop
+#endif
+
+#include <any>
+#include <unordered_map>
+
```

Comment:
Needs a big comment for future readers with a 101
- what is a failpoint
- how to use it from code side
- how to use it in tests.